tools/metrics/gauges.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+)
+
+func QuotaGauge(c *Client, interval time.Duration) MetricReadCloser {
+	return NewMetricPoller(interval, func(w MetricWriter) error {
+		orgs, err := c.cf.ListOrgs()
+		if err != nil {
+			return err
+		}
+
+		allocatedMemoryByOrgs := 0
+		allocatedMemoryByApps := 0
+		allocatedServices := 0
+		allocatedRoutes := 0
+
+		for _, org := range orgs {
+			quota, err := org.Quota()
+			if err != nil {
+				return err
+			}
+			allocatedMemoryByOrgs += quota.MemoryLimit
+			allocatedServices += quota.TotalServices
+			allocatedRoutes += quota.TotalRoutes
+		}
+
+		apps, err := c.cf.ListApps()
+		if err != nil {
+			return err
+		}
+		for _, app := range apps {
+			allocatedMemoryByApps += (app.Memory * app.Instances)
+		}
+		return w.WriteMetrics([]Metric{
+			{
+				Kind:  Gauge,
+				Time:  time.Now(),
+				Name:  "quota.services.reserved", // number of services reserved by quotas
+				Value: float64(allocatedServices),
+			},
+			{
+				Kind:  Gauge,
+				Time:  time.Now(),
+				Name:  "quota.memory.reserved", // memory reserved by org quotas
+				Value: float64(allocatedMemoryByOrgs),
+			},
+			{
+				Kind:  Gauge,
+				Time:  time.Now(),
+				Name:  "quota.memory.allocated", // memory reserved by apps
+				Value: float64(allocatedMemoryByApps),
+			},
+		})
+	})
+}
+
+func UserCountGauge(c *Client, interval time.Duration) MetricReadCloser {
+	return NewMetricPoller(interval, func(w MetricWriter) error {
+		// global auditor role cannot use /v2/users
+		// so we have to fetch users from each org
+		orgs, err := c.cf.ListOrgs()
+		if err != nil {
+			return err
+		}
+
+		numberOfUsers := 0
+
+		for _, org := range orgs {
+			var response struct {
+				TotalResults int `json:"total_results"`
+			}
+			if err := c.get(fmt.Sprintf("/v2/organizations/%s/users", org.Guid), &response); err != nil {
+				return err
+			}
+			numberOfUsers += response.TotalResults
+		}
+		return w.WriteMetrics([]Metric{
+			{
+				Kind:  Gauge,
+				Time:  time.Now(),
+				Name:  "users.count",
+				Value: float64(numberOfUsers),
```

Comment:
Yep you're right ... this way of calculating users was a quick fix for the fact the global auditor role could not fetch `/v2/users`... clearly _too_ quick.

I've de-duplicated it now by counting unqiue user guids.